src/app/modules/main/module.nim

```diff
@@ -43,6 +43,11 @@ export io_interface
 type 
   Module*[T: io_interface.DelegateInterface] = ref object of io_interface.AccessInterface
     delegate: T
+    events: EventEmitter
```

Comment:
@jrainville all these instances, starting from this line below, you already have in `controller` of this module, also as I can see the only reason why you added them to be private members of this class is to use them in `communityJoined` which is by the way called from `controller`, if you don't mind please remove them and call `communityJoined` with necessary parameters from the controller.

A reason why I am asking that, just cause tomorrow we may have some lazy dev here which will do some service calls from a module part (cause he may see that we already have services as private members of a module) and that way we may start mixing business logic (which should be in module) and service providers logic (which should be part of the controller responsibility) and corrupt that way the rules we set for our architecture.